docs/content/guide/component.ngdoc

```diff
@@ -0,0 +1,453 @@
+    bindings: {user: '='}
+  });
+  myMod.config(function($routeProvider) {
+    $routeProvider.when('/', {
+      template: '<home user="$resolve.user"></home>',
+      resolve: {user: function($http) { return $http.get('...'); }}
+    });
+  });
+```
+
+## Intercomponent Communication
+
+Directives can require the controllers of other directives to enable communication
+between the directives. This can be achieved in a component by providing an
+object mapping for the `require` property.  Here is a tab pane example built
+from components:
+
+<example module="docsTabsExample">
+  <file name="script.js">
+angular.module('docsTabsExample', [])
+  .component('myTabs', {
+    transclude: true,
+    controller: function() {
+      var panes = this.panes = [];
+      this.select = function(pane) {
+        angular.forEach(panes, function(pane) {
+          pane.selected = false;
+        });
+        pane.selected = true;
+      };
+      this.addPane = function(pane) {
+        if (panes.length === 0) {
+          this.select(pane);
+        }
+        panes.push(pane);
+      };
+    },
+    templateUrl: 'my-tabs.html'
+  })
+  .component('myPane', {
+    transclude: true,
+    require: {tabsCtrl: '^myTabs'},
+    bindings: {
+      title: '@'
+    },
+    controller: function() {
+      this.$onInit = function() {
+        this.tabsCtrl.addPane(this);
+        console.log(this);
+      };
+    },
+    templateUrl: 'my-pane.html'
+  });
+  </file>
+  <file name="index.html">
+  <my-tabs>
+    <my-pane title="Hello">
+      <h4>Hello</h4>
+      <p>Lorem ipsum dolor sit amet</p>
+    </my-pane>
+    <my-pane title="World">
+      <h4>World</h4>
+      <em>Mauris elementum elementum enim at suscipit.</em>
+      <p><a href ng-click="i = i + 1">counter: {{i || 0}}</a></p>
+    </my-pane>
+  </my-tabs>
+  </file>
+  <file name="my-tabs.html">
+  <div class="tabbable">
+    <ul class="nav nav-tabs">
+      <li ng-repeat="pane in $ctrl.panes" ng-class="{active:pane.selected}">
+        <a href="" ng-click="$ctrl.select(pane)">{{pane.title}}</a>
+      </li>
+    </ul>
+    <div class="tab-content" ng-transclude></div>
+  </div>
+  </file>
+  <file name="my-pane.html">
+    <div class="tab-pane" ng-show="$ctrl.selected" ng-transclude></div>
+  </file>
+</example>
+
+
+# Unit-testing Components
+
+The easiest way to unit-test a component is by using the {@link ngMock.$componentController $componentController}
+that is included in {@link ngMock}. The advantage of this method is that you do not have
+to create any DOM elements. The following example shows how to the test the `heroDetail` component
+from above.
+
+The examples use the [Jasmine](http://jasmine.github.io/) testing framework.
+
+**Controller Test:**
+```js
+describe('component: heroDetail', function() {
+  var component, scope, hero;
+
+  beforeEach(module('simpleComponent'));
+
+  beforeEach(inject(function($rootScope, $componentController) {
+    scope = $rootScope.$new();
```

Comment:
But it's possible that a controller uses the injected scope, even though we don't show this here,